File: C++/565. Array Nesting.cpp
class Solution {
public:
    int arrayNesting(vector<int>& nums) {
        vector<int> vis(nums.size(), 0);
        int ans = 0;
        for(int i=0;i<nums.size();i++)
        {
            int j = nums[i];
            int tmp = 0;
            while(!vis[j])
            {
                vis[j] = 1;
                j = nums[j];
                tmp += 1;
            }
            ans = max(ans, tmp);
        }
        return ans;
    }
};

File: Python/565. Array Nesting.py
class Solution:
    def arrayNesting(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        vis = [0]*len(nums)
        ans = 0
        for i in range(len(nums)):
            num = nums[i]
            tmp = 0
            while vis[num] != 1:
                vis[num] = 1
                num = nums[num]
                tmp += 1
            ans = max(ans, tmp)
        return ans